src/frontends/tensorflow/src/input_model.cpp

```diff
@@ -171,6 +171,7 @@ std::vector<std::shared_ptr<OpPlace>> InputModel::InputModelTFImpl::get_op_place
 std::vector<std::shared_ptr<OpPlace>> InputModel::InputModelTFImpl::determine_cut_nodes() const {
     std::vector<std::shared_ptr<OpPlace>> topologically_sorted_ops;
     std::stack<std::shared_ptr<OpPlace>> ops_to_do;
```

Comment:
nitpick: I would recommend storing OpPlace* in all these data structures because in some cases storing shared_ptr<OpPlace> can cause to unexpected issues e.g.
![image](https://user-images.githubusercontent.com/21181928/202680979-08a4cf20-1f98-4e45-a825-7bd72b371459.png)

Reply:
I use unordered_set separately to understand if some pointer to Node is being processed. Not stack.

Reply:
For `unordered_set` these two pointers will not be added twice.